### PR TITLE
Split up bazel test into tune / non tune tests

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This should avoid running python tests when tune lib is changed only.

### CODE REVIEW

**Comment by richardliaw** [comment]
This looks good to me (re tune tests); are there other tests in `python/ray/` that are not covered by `python/ray/tests` and `python/ray/tune` and `python/ray/experimental/serve`?

**Comment by richardliaw** [comment]
btw, test failed

File: .travis.yml [hunk 1]
```diff
@@ -169,11 +169,13 @@ script:
...
   - cd ../..
 
-  # bazel python tests. This should be run last to keep its logs at the end of travis logs.
+  # random python tests TODO(ekl): these should be moved to bazel
   - if [ $RAY_CI_PYTHON_AFFECTED == "1" ]; then RAY_FORCE_DIRECT=0 python -m pytest -v --durations=5 --timeout=300 python/ray/experimental/test/async_test.py; fi
   - if [ $RAY_CI_PYTHON_AFFECTED == "1" ]; then python -m pytest -v --durations=5 --timeout=300 python/ray/tests/py3_test.py; fi
 
-  - ./ci/keep_alive bazel test --spawn_strategy=local --flaky_test_attempts=3 --nocache_test_results --test_verbose_timeout_warnings --progress_report_interval=100 --show_progress_rate_limit=100 --show_timestamps --test_output=errors --test_tag_filters=-jenkins_only python/ray/...
+  # bazel python tests. This should be run last to keep its logs at the end of travis logs.
+  - if [ $RAY_CI_PYTHON_AFFECTED == "1" ]; ./ci/keep_alive bazel test --spawn_strategy=local --flaky_test_attempts=3 --nocache_test_results --test_verbose_timeout_warnings --progress_report_interval=100 --show_progress_rate_limit=100 --show_timestamps --test_output=errors --test_tag_filters=-jenkins_only python/ray/tests/...; fi
```
Comment (simon-mo) [inline .travis.yml]: Should python run last?

**Review by simon-mo — APPROVED** [review]

**Comment by richardliaw** [comment]
cc @ericl tests failing

**Pull request merged**